elixir/lib/sqlite_vss.ex

```diff
@@ -0,0 +1,245 @@
+       :binary
+     )
+   ),
+   order_by: [asc: v0.distance],
+   limit: ^100,
+   select: [:rowid, :distance]),
+   on: v1.rowid == p0.id>
+
+  """
+  @spec search_query(Ecto.Query.t(), String.t(), String.t(), binary(), integer() | none()) :: Ecto.Query.t()
+  def search(query, table_name, embedding_column_name, vector, limit \\ 100) do
+    subquery =
+      from(s in table_name,
+        where:
+          fragment(
+            "vss_search(?, ?)",
+            field(s, ^String.to_existing_atom(embedding_column_name)),
+            type(^vector, :binary)
+          ),
+        select: [:rowid, :distance],
+        order_by: s.distance,
+        limit: ^limit
+      )
+
+    from(p in query,
+      join: s in subquery(subquery),
+      on: s.rowid == p.id
+    )
+  end
+
+  @doc false
+  def start(_, _) do
+    unless Application.get_env(:sqlite_vss, :version) do
+      Logger.warn("""
+      sqlite_vss version is not configured. Please set it in your config files:
+
+          config :sqlite_vss, :version, "#{latest_version()}"
+      """)
+    end
+
+    Supervisor.start_link([], strategy: :one_for_one)
+  end
+
+  @doc """
+  Returns the configured sqlite_vss version.
+  """
+  def configured_version do
```

Comment:
Instead of having a separate `:version` supplied as config, can we instead use the version of the current package? All releases of `sqlite-vss` and the Python/Node/Deno bindings are all released as the same time. So a new `v0.1.1` version of the SQL API will also correspond with the `v0.1.1` versions of the PyPi package, NPM package, and `deno.land/x` module.

If someone wants to bump their version of their `sqlite_vss` package, I'd like them to just use `mix` to upgrade the package, and not have a separate config for that